### ISSUE TITLE
Target cluster endpoint is unreachable when we do make clean and make again due to firewall rules

### ISSUE THREAD

**Issue by kashifest** [metal3-io/metal3-dev-env#298]
When we do a `make clean` and `make` again and try to do the provisioning of the control-plane again we see the following error in logs of CAPI and CAKCP:
```
"failed to create remote cluster client: failed to create client for workload cluster metal3/test1: Get https://192.168.111.249:6443/api?timeout=30s: dial tcp 192.168.111.249:6443: connect: connection refused" "cluster"="test1" "kubeadmControlPlane"="test1-controlplane" "namespace"="metal3" 
```
This particularly happens when we have Kind as the ephemeral cluster and we have done a `make clean` and `make` again. The iptable rules are as follows:

```
Chain FORWARD (policy DROP 14 packets, 560 bytes)
 pkts bytes target     prot opt in     out     source               destination         
 272K 3441M ACCEPT     all  --  *      baremetal  0.0.0.0/0            192.168.111.0/24     ctstate RELATED,ESTABLISHED
 106K 6508K ACCEPT     all  --  baremetal *       192.168.111.0/24     0.0.0.0/0           
   16  5236 ACCEPT     all  --  baremetal baremetal  0.0.0.0/0            0.0.0.0/0           
 9177  551K REJECT     all  --  *      baremetal  0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable
    0     0 REJECT     all  --  baremetal *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable
    0     0 ACCEPT     all  --  *      virbr0  0.0.0.0/0            192.168.122.0/24     ctstate RELATED,ESTABLISHED
    0     0 ACCEPT     all  --  virbr0 *       192.168.122.0/24     0.0.0.0/0           
    0     0 ACCEPT     all  --  virbr0 virbr0  0.0.0.0/0            0.0.0.0/0           
    0     0 REJECT     all  --  *      virbr0  0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable
    0     0 REJECT     all  --  virbr0 *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable
 108K 1068M DOCKER-USER  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 108K 1068M DOCKER-ISOLATION-STAGE-1  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
59814 1065M ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
47362 3492K ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           
89323 2014M ACCEPT     all  --  *      baremetal  0.0.0.0/0            192.168.111.0/24    
    0     0 ACCEPT     all  --  provisioning *       0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
    0     0 ACCEPT     tcp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            tcp dpt:6180
    0     0 ACCEPT     tcp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            tcp dpt:5050
    0     0 ACCEPT     tcp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            tcp dpt:6385
    0     0 ACCEPT     tcp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            tcp dpt:9999
   72  6408 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:5353
   73 27966 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:67
    0     0 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:68
    0     0 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:546
    0     0 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:547
    0     0 ACCEPT     udp  --  provisioning *       0.0.0.0/0            0.0.0.0/0            udp dpt:69
21442  858K ACCEPT     112  --  provisioning *       0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     icmp --  provisioning *       0.0.0.0/0            0.0.0.0/0           
```

As can be seen the REJECT rule for `baremetal` comes now in the beginning before the ACCEPT and as a result it rejects the networking towards baremetal network. One solution could be to change the firewall rules so that the rules are made absent while we do `make clean` and they are made present again during `make` so that we always have the order of the rules correct.

**Comment by maelk** [comment]
This should be fixed by #311 @kashifest can you confirm ?

**Comment by kashifest** [comment]
Yes I guess. we can close this issue
